### ISSUE TITLE
sealed constructors

### ISSUE THREAD

**Issue by gavinking** [ceylon/ceylon-compiler#2018]
A `sealed` constructor currently results in:

```
Annotation type not applicable to this kind of declaration
```

**Comment by tombentley** [comment]
Well if you were using the regular `sealed` then its annotation class is supposed to be constrained to a `ClassOrInterfaceDeclaration`, so the typechecker should produce an error when it's used on a constructor, no?

And if I modify `SealedAnnotation` to be constrained to `ClassOrInterfaceDeclaration|ConstructorDeclaration` then it works for me.

**Comment by gavinking** [comment]
@tombentley ah ok, turns out i wasn't checking annotation constraints for constructors.

**Comment by tombentley** [comment]
Do you want a spec issue for this?
